bundles-lib/src/main/java/org/apache/metron/bundles/ExtensionManager.java

```diff
@@ -0,0 +1,436 @@
+   * Loads all extension class types that can be found on the bootstrap classloader and by creating
+   * classloaders for all BUNDLES found within the classpath.
+   *
+   * @param bundles the bundles to scan through in search of extensions
+   */
+  public static void discoverExtensions(final Bundle systemBundle, final Set<Bundle> bundles)
+      throws NotInitializedException {
+    checkInitialized();
+    // get the current context class loader
+    ClassLoader currentContextClassLoader = Thread.currentThread().getContextClassLoader();
+
+    // load the system bundle first so that any extensions found in JARs directly in lib will be registered as
+    // being from the system bundle and not from all the other Bundles
+    loadExtensions(systemBundle);
+    bundleCoordinateBundleLookup.put(systemBundle.getBundleDetails().getCoordinates(), systemBundle);
+
+    // consider each bundle class loader
+    for (final Bundle bundle : bundles) {
+      // Must set the context class loader to the bundle classloader itself
+      // so that static initialization techniques that depend on the context class loader will work properly
+      final ClassLoader ncl = bundle.getClassLoader();
+      Thread.currentThread().setContextClassLoader(ncl);
+      loadExtensions(bundle);
+
+      // Create a look-up from withCoordinates to bundle
+      bundleCoordinateBundleLookup.put(bundle.getBundleDetails().getCoordinates(), bundle);
+    }
+
+    // restore the current context class loader if appropriate
+    if (currentContextClassLoader != null) {
+      Thread.currentThread().setContextClassLoader(currentContextClassLoader);
+    }
+  }
+
+  /**
+   * Returns a bundle representing the system class loader.
+   *
+   * @param bundleProperties a BundleProperties instance which will be used to obtain the default
+   * Bundle library path, which will become the working directory of the returned bundle
+   * @return a bundle for the system class loader
+   */
+  public static Bundle createSystemBundle(final FileSystemManager fileSystemManager,
+      final BundleProperties bundleProperties) throws FileSystemException, URISyntaxException {
+    final ClassLoader systemClassLoader = ClassLoader.getSystemClassLoader();
+
+    final String bundlesLibraryDirectory = bundleProperties
+        .getProperty(BundleProperties.BUNDLE_LIBRARY_DIRECTORY);
+    if (StringUtils.isBlank(bundlesLibraryDirectory)) {
+      throw new IllegalStateException(
+          "Unable to create system bundle because " + BundleProperties.BUNDLE_LIBRARY_DIRECTORY
+              + " was null or empty");
+    }
+    final URI bundlesLibraryDirURI = bundleProperties.getBundleLibraryDirectory();
+    FileObject bundleDir = fileSystemManager.resolveFile(bundlesLibraryDirURI);
+
+    // Test if the source Bundles can be read
+    FileUtils.ensureDirectoryExistAndCanRead(bundleDir);
+
+    final BundleDetails systemBundleDetails = new BundleDetails.Builder()
+        .withBundleFile(bundleDir)
```

Comment:
I did two things:
1. create a FileObject implementation, a dummy to use here
2. added a comment